### ISSUE TITLE
Undeclared variable in complex_decode_double() function

### ISSUE THREAD

**Issue by japinli** [greenplum-db/gpdb#5879]
I compile Greenplumn 6.0.0-alpha.0+dev.11215.gabccfe612b build dev on SunOS and get the following error:

```
gcc -D_POSIX_PTHREAD_SEMANTICS -std=gnu99  -Wall -Wmissing-prototypes -Wpointer-arith -Wendif-labels -Wmissing-format-attribute -Wformat-security -fno-strict-aliasing -fwrapv -fexcess-precision=standard -fno-aggressive-loop-optimizations -Wno-unused-but-set-variable -Wno-address -I/opt/local/include -Werror=uninitialized -Werror=implicit-function-declaration -I../../../../src/interface
s/libpq -I../../../../src/include -I/opt/local/include   -c -o complex_type.o complex_type.c
complex_type.c: In function 'complex_decode_double':
complex_type.c:121:7: error: 'endptr' undeclared (first use in this function)
   if (endptr != ptr_iter && '\0' == endptr[-1])
       ^
complex_type.c:121:7: note: each undeclared identifier is reported only once for each function it appears in
<builtin>: recipe for target 'complex_type.o' failed
```

Then, I checked the file of src/backend/utils/adt/complex_type.c, and find the following codes:

```
#ifdef HAVE_BUGGY_SOLARIS_STRTOD
	else
	{
		/*
		 * Many versions of Solaris have a bug wherein strtod sets endptr to
		 * point one byte beyond the end of the string when gvin "inf" or
		 * "infinity".
		 */
		if (endptr != ptr_iter && '\0' == endptr[-1])
		{
			end_ptr--;
		}
	}
#endif   /* HAVE_BUGGY_SOLARIS_STRTOD */
```

In the above codes, I can not find `endptr` variable declaration in the `complex_decode_double()` function which includes the above codes. Is it `endptr` variable written by mistake?

**Comment by hlinnaka** [comment]
This code was originally copy-pasted from `float8in`, and modified. One of the modifications was to rename `endptr` to `end_ptr`, but this reference in the #ifdef block was missed.

I opened PR #6536 to fix this. Please take a look, and thanks for the report!

**Comment by hlinnaka** [comment]
Fixed by commit c77e2c9bf0. Thanks for the report, @japinli!
